In [8]:
import ultralytics
from ultralytics import YOLO

In [9]:
ultralytics.checks()

Ultralytics 8.3.63 🚀 Python-3.10.16 torch-2.5.1 CPU (Apple M3)
Setup complete ✅ (8 CPUs, 16.0 GB RAM, 132.2/460.4 GB disk)


In [10]:
model = YOLO('yolo11n.pt')  # 'n' denotes the nano version

In [11]:
model.device

device(type='cpu')

In [ ]:
model.train(
    data='dataset.yaml',  # Path to the dataset configuration file
    epochs=50,            # Number of training epochs
    imgsz=640,            # Image size for training
    batch=16,             # Batch size (adjust based on your hardware capabilities)
    name='drone_vs_bird_finetune',  # Name of the training run
    pretrained=True,      # Use pre-trained weights
    device='cpu',         # GPU device index (set to 'cpu' on mac, GPU acceration is only for CUDA)
    amp=True              # Enable Automatic Mixed Precision (for faster training)      
)


New https://pypi.org/project/ultralytics/8.3.64 available 😃 Update with 'pip install -U ultralytics'
Ultralytics 8.3.63 🚀 Python-3.10.16 torch-2.5.1 CPU (Apple M3)
engine/trainer: task=detect, mode=train, model=yolo11n.pt, data=dataset.yaml, epochs=50, time=None, patience=100, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=cpu, workers=8, project=None, name=drone_vs_bird_finetune7, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=

train: Scanning /Users/aditya/Documents/Github/drone-vs-bird-challenge/dataset/labels/train... 68168 images, 0 backgrounds, 5963 corrupt: 100%|██████████| 68168/68168 [00:10<00:00, 6523.68it/s]

train: WARNING ⚠️ /Users/aditya/Documents/Github/drone-vs-bird-challenge/dataset/images/train/2019_08_19_C0001_5319_phantom_0475.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0112]
train: WARNING ⚠️ /Users/aditya/Documents/Github/drone-vs-bird-challenge/dataset/images/train/2019_08_19_C0001_5319_phantom_0477.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0268]
train: WARNING ⚠️ /Users/aditya/Documents/Github/drone-vs-bird-challenge/dataset/images/train/2019_08_19_C0001_5319_phantom_0478.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0341]
train: WARNING ⚠️ /Users/aditya/Documents/Github/drone-vs-bird-challenge/dataset/images/train/2019_08_19_C0001_5319_phantom_0480.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0497]
train: WARNING ⚠️ /Users/aditya/Documents/Github/drone-vs-bird-challenge/dataset/images/train/2019_08_19_C0001_5

train: New cache created: /Users/aditya/Documents/Github/drone-vs-bird-challenge/dataset/labels/train.cache


val: Scanning /Users/aditya/Documents/Github/drone-vs-bird-challenge/dataset/labels/val... 9739 images, 0 backgrounds, 891 corrupt: 100%|██████████| 9739/9739 [00:01<00:00, 6514.97it/s]

val: WARNING ⚠️ /Users/aditya/Documents/Github/drone-vs-bird-challenge/dataset/images/val/2019_08_19_C0001_5319_phantom_0479.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.0419]
val: WARNING ⚠️ /Users/aditya/Documents/Github/drone-vs-bird-challenge/dataset/images/val/2019_08_19_C0001_5319_phantom_0520.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.3273]
val: WARNING ⚠️ /Users/aditya/Documents/Github/drone-vs-bird-challenge/dataset/images/val/2019_08_19_C0001_5319_phantom_0522.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.3404]
val: WARNING ⚠️ /Users/aditya/Documents/Github/drone-vs-bird-challenge/dataset/images/val/2019_08_19_C0001_5319_phantom_0532.jpg: ignoring corrupt image/label: non-normalized or out of bounds coordinates [     1.4164]
val: WARNING ⚠️ /Users/aditya/Documents/Github/drone-vs-bird-challenge/dataset/images/val/2019_08_19_C0001_5319_phantom_0541.jpg

val: New cache created: /Users/aditya/Documents/Github/drone-vs-bird-challenge/dataset/labels/val.cache
Plotting labels to runs/detect/drone_vs_bird_finetune7/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: SGD(lr=0.01, momentum=0.9) with parameter groups 81 weight(decay=0.0), 88 weight(decay=0.0005), 87 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 0 dataloader workers
Logging results to runs/detect/drone_vs_bird_finetune7
Starting training for 50 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


       1/50         0G      2.945      14.92      1.522         31        640:   1%|          | 37/3888 [04:44<9:58:46,  9.33s/it]

Ultralytics 8.3.63 🚀 Python-3.10.16 torch-2.5.1 CPU (Apple M3)
engine/trainer: task=detect, mode=train, model=yolo11n.pt, data=dataset.yaml, epochs=50, time=None, patience=100, batch=16, imgsz=640, save=True, save_period=-1, cache=False, device=cpu, workers=8, project=None, name=drone_vs_bird_finetune, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=True, opset=None, workspace=None, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, bgr=0.0, mosaic=1.0, mixup=0.0, copy_paste=0.0, copy_paste_mode=flip, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=runs/detect/drone_vs_bird_finetune